app/recipe/serializers.py

```diff
@@ -0,0 +1,38 @@
+from rest_framework import serializers
+
+from core.models import Ingredient, Recipe
+
+
+class IngredientSerializer(serializers.ModelSerializer):
+    """Serializer for ingredient objects"""
+
+    class Meta:
+        model = Ingredient
+        fields = ('name',)
+        read_only_fields = ('id',)
+
+
+class RecipeSerializer(serializers.ModelSerializer):
+    """Serializer for recipe objects"""
+    ingredients = IngredientSerializer(many=True)
+
+    def create(self, validated_data):
+        ingredients = validated_data.pop('ingredients')
+        recipe = Recipe.objects.create(**validated_data)
+        for ingredient in ingredients:
+            Ingredient.objects.create(recipe=recipe, **ingredient)
+        return recipe
+
+    def update(self, instance, validated_data):
+        instance.name = validated_data.get('name', instance.name)
+        Ingredient.objects.filter(recipe=instance.id).delete()
```

Comment:
What happens, if the ingredient is used in another recipe?

Reply:
Ingredients can have duplicate names, e.g. we can have another ingredient called `cheese` in `Cheeseburger` that is a different database record to the `cheese` in `Pizza`. But I think it would be a good test to prove that this is the case, by adding another recipe called `Cheeseburger` that also has `cheese` in it, and proving that the `cheese` in `Cheeseburger` is not deleted when we remove the `cheese` from the `Pizza`. So I will add that extra bit to the test now.